lib/pushgateway.js

```diff
@@ -75,7 +75,8 @@ function useGateway(method, job, groupings, callback) {
 	});
 
 	if (method !== 'DELETE') {
-		req.write(this.registry.metrics());
+		const metrics = await this.registry.metrics();
+		req.write(metrics);
```

Comment:
I'd rather not mix a callback with async/await, although you could do it if you add a `try/catch` block here and pass the caught exception to the callback. Coincidentally, I started working on this issue a bit before you opened this PR. This is what I had so far using Promise/then/catch:

```diff
        if (method !== 'DELETE') {
-               req.write(this.registry.metrics());
+               this.registry
+                       .metrics()
+                       .then(metrics => {
+                               req.write(metrics);
+                               req.end();
+                       })
+                       .catch(err => {
+                               callback(err);
+                       });
+       } else {
+               req.end();
        }
-       req.end();
```

(I was trying to see if I could easily write a test for it, so didn't open a PR.)

Reply:
It's ok to use promise instead of async/await.
I have updated mr.